### PR TITLE
Issue 1053/placeholder organizations landing page

### LINKED ISSUE #1053: Placeholder organizer landing page
## Description
A user should be able to go to a single page and find links to all of their organizations. At the moment there is no such page, and while it will take a long time to design and build a final version, we need to have something for the MVP that will be used in a couple of weeks.

## Relevant Job Stories
None

## Prerequisites
None

## Requirements
- [x] Create a page at /organize
- [x] List all of the user's organizations there
- [x] Redirect /organize/ID to /organize/ID/campaigns

## Possible implementations
None

## Design specifications
There is no design. Let's just create something very basic. Here is the screen from the 2020 Christmas Wireframes:

image

## Open questions
None

### PR DESCRIPTION
**Pull request by rebecarubio**
## Description
This PR adds a landing page that displays all the organizations (and sub-orgs) that the user belongs to. When the user selects a organizations it´s redirected to the campaigns page of the organization ( /organize/ID/campaigns)

## Screenshots
image

## Changes
* Adds `OrganizationsList `component
* Adds organize `index `page to display the `OrganizationsList `component
* Adds redux logic to get organizations data from user (added `model`, `repo `and `store `organizations files)
* Adds it correspondent `messageIds `file to use localization in Organizations component/page 

## Notes to reviewer
There´s no breadcrumbs available because we don´t have a org selected. But if you think we need to be introduced I can do it!

## Related issues
Resolves #1053

### CODE REVIEW

File: src/features/organizations/components/OrganizationsList.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      {(data) => {
+        return (
+          <Box style={{ margin: '30px' }}>
+            <Typography variant="h3">{messages.page.title()}</Typography>
+            {data.map((org: ZetkinOrganization) => {
+              return (
+                <Box key={org.id} style={{ marginTop: '15px' }}>
+                  <ZUIList>
+                    <NextLink href={`/organize/${org.id}/campaigns/`} passHref>
+                      <Link underline="hover">{org.title}</Link>
```
Comment (richardolsson) [inline src/features/organizations/components/OrganizationsList.tsx:28]: I think the organization avatar should also be included here. You should be able to load it from `/api/orgs/ID/avatar`.

File: src/features/organizations/l10n/messageIds.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { m, makeMessages } from 'core/i18n';
+
+export default makeMessages('feat.organizations', {
+  page: {
+    title: m('Select organizations:'),
```
Comment (richardolsson) [inline src/features/organizations/l10n/messageIds.ts:5]: I think this should be `Select organization`.

File: src/features/organizations/repos/OrganizationsRepo.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+export default class OrganizationsRepo {
+  private _apiClient: IApiClient;
+  private _store: Store;
+
+  constructor(env: Environment) {
+    this._store = env.store;
+    this._apiClient = env.apiClient;
+  }
+
+  getOrganizations(): IFuture<ZetkinOrganization[]> {
```
Comment (richardolsson) [inline src/features/organizations/repos/OrganizationsRepo.ts:18]: I think this should be called `getUserOrganizations()`.

File: src/features/organizations/store.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { ZetkinOrganization } from 'utils/types/zetkin';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { remoteList, RemoteList } from 'utils/storeUtils';
+
+export interface OrganizationsStoreSlice {
+  allOrgs: RemoteList<ZetkinOrganization>;
```
Comment (richardolsson) [inline src/features/organizations/store.ts:6]: I think this should be called `userOrgList`.

File: src/features/organizations/components/OrganizationsList.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    <ZUIFuture future={model.getData()}>
+      {(data) => {
+        return (
+          <Box style={{ margin: '30px' }}>
+            <Typography variant="h3">{messages.page.title()}</Typography>
+            {data.map((org: ZetkinOrganization) => {
+              return (
+                <Box key={org.id} style={{ marginTop: '15px' }}>
+                  <ZUIList>
+                    <NextLink href={`/organize/${org.id}/campaigns/`} passHref>
```
Comment (richardolsson) [inline src/features/organizations/components/OrganizationsList.tsx:27]: I don't think this should link to `/organize/ID/campaigns`. I think it should link to `/organize/ID` (and that URL should be configured in `next.config.js` to redirect to campaigns).

**Review by richardolsson — CHANGES_REQUESTED** [review]
Awesome work! I can see that you learned a lot. I'm glad. 😊 

I would like some minor changes to the code, but nothing big or structural.

File: src/pages/organize/index.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import DefaultLayout from 'utils/layout/DefaultLayout';
+import Head from 'next/head';
+import messageIds from 'features/organizations/l10n/messageIds';
+import OrganizationsDataModel from 'features/organizations/models/OrganizationsDataModel';
+import { PageWithLayout } from 'utils/types';
+import { useMessages } from 'core/i18n';
+import useModel from 'core/useModel';
+import UserOrganizations from 'features/organizations/components/OrganizationsList';
+
```
Comment (richardolsson) [inline src/pages/organize/index.tsx:9]: One more thing! You also need a `getServerSideProps()` here to check that the user is authenticated.

Not having that here also seems to be causing build errors, which in turn causes the playwright test suite to fail (or not even run).

**Comment by richardolsson** [comment]
Finally, maybe @Bagera can recommend some simple tweaks to the visuals of the page?

File: src/features/organizations/l10n/messageIds.ts (hunk 2 above)
Comment (rebecarubio) [reply src/features/organizations/l10n/messageIds.ts:5]: Right!

File: src/features/organizations/components/OrganizationsList.tsx (hunk 1 above)
Comment (rebecarubio) [reply src/features/organizations/components/OrganizationsList.tsx:28]: Cool!

File: src/features/organizations/components/OrganizationsList.tsx (hunk 5 above)
Comment (rebecarubio) [reply src/features/organizations/components/OrganizationsList.tsx:27]: Got it! ;)

File: src/pages/organize/index.tsx (hunk 6 above)
Comment (rebecarubio) [reply src/pages/organize/index.tsx:9]: Of course! I'll add it! thanks!

**Review by richardolsson — APPROVED** [review]
Perfect! 💯 

I think we might still need some designer eyes on this, but we can revisit the design next week if we find time.

**Comment by Bagera** [comment]
> I think we might still need some designer eyes on this, but we can revisit the design next week if we find time.

I agree, let's look at it at K&K next week

**Pull request merged**